test/functional/tests/fault_injection/test_fault_injection_many_to_one.py

```diff
@@ -0,0 +1,343 @@
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core1", DiskTypeLowerThan("cache"))
+@pytest.mark.require_disk("core2", DiskTypeLowerThan("cache"))
+def test_one_core_fail(cache_mode):
+    """
+        title: Test if OpenCAS correctly handles failure of one of multiple core devices.
+        description: |
+          When one core device fails in a single cache instance all blocks previously occupied
+          should be available to other core devices.
+          Test is without pass through mode.
+        pass_criteria:
+          - No system crash.
+          - Second core is able to use OpenCAS.
+    """
+    with TestRun.step("Prepare one cache and two core devices."):
+        cache_dev = TestRun.disks['cache']
+        cache_dev.create_partitions([Size(1, Unit.GibiByte)])
+        cache_part = cache_dev.partitions[0]
+        core_dev1 = TestRun.disks['core1']  # This device would be unplugged.
+        core_dev1.create_partitions([Size(2, Unit.GibiByte)])
+        core_part1 = core_dev1.partitions[0]
+        core_dev2 = TestRun.disks['core2']
+        core_dev2.create_partitions([Size(2, Unit.GibiByte)])
+        core_part2 = core_dev2.partitions[0]
+        Udev.disable()
+
+    with TestRun.step("Start cache"):
+        cache = casadm.start_cache(cache_part, cache_mode, force=True)
+        caches_count = len(casadm_parser.get_caches())
+        if caches_count != 1:
+            TestRun.fail(f"Expected caches count: 1; Actual caches count: {caches_count}.")
+
+    with TestRun.step("Add both core devices to cache."):
+        core1 = cache.add_core(core_part1)
+        core2 = cache.add_core(core_part2)
+        cores_count = len(casadm_parser.get_cores(cache.cache_id))
+        if cores_count != 2:
+            TestRun.fail(f"Expected cores count: 2; Actual cores count: {cores_count}.")
+
+    with TestRun.step("Change sequential cutoff policy."):
+        cache.set_seq_cutoff_policy(SeqCutOffPolicy.never)
+
+    with TestRun.step("Fill cache with pages from the first core."):
+        blocks = int(cache.size.value / block_size.value)
+        if CacheMode.get_traits(cache_mode) == CacheModeTrait.InsertRead:
+            dd = (Dd()
+                  .input(core1.system_path)
+                  .output("/dev/null")
+                  .block_size(block_size)
+                  .count(blocks))
+        elif CacheMode.get_traits(cache_mode) != CacheModeTrait.InsertRead:
+            dd = (Dd()
+                  .input("/dev/urandom")
+                  .output(core1.system_path)
+                  .block_size(block_size)
+                  .count(blocks))
+        dd.run()
+        cache_occupied_blocks_before = cache.get_occupancy()
+
+    with TestRun.step("Unplug the first core device."):
+        core_dev1.unplug()
+
+    with TestRun.step("Check if core device is really out of cache."):
+        try:
+            cache.get_core_devices()
```

Comment:
this method will return the other core and it won't raise any exception after unplugging first core. Instead you can get list of cores and check if unplugged core is not listed there

Reply:
Changed checking that and verified my method for doing it.
Also this works too, but this method isn't clear and readable.